File: src/main/java/io/github/takenoha/towerdefense/paper/TowerManagementInventoryHolder.java
package io.github.takenoha.towerdefense.paper;

import java.util.Objects;
import java.util.UUID;
import org.bukkit.inventory.Inventory;
import org.bukkit.inventory.InventoryHolder;

/** Identifies the tower-management inventory and its target tower. */
public final class TowerManagementInventoryHolder implements InventoryHolder {
    private final UUID towerId;
    private Inventory inventory;

    public TowerManagementInventoryHolder(UUID towerId) {
        this.towerId = Objects.requireNonNull(towerId, "towerId");
    }

    public UUID towerId() {
        return towerId;
    }

    void attach(Inventory inventory) {
        this.inventory = Objects.requireNonNull(inventory, "inventory");
    }

    @Override
    public Inventory getInventory() {
        if (inventory == null) {
            throw new IllegalStateException("the GUI inventory has not been attached");
        }
        return inventory;
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/paper/TowerManagementInventoryHolder.kt
package io.github.takenoha.towerdefense.paper

import java.util.Objects
import java.util.UUID
import org.bukkit.inventory.Inventory
import org.bukkit.inventory.InventoryHolder

/** Identifies the tower-management inventory and its target tower. */
class TowerManagementInventoryHolder(towerId: UUID) : InventoryHolder {
    private val towerIdValue: UUID = Objects.requireNonNull(towerId, "towerId")
    private var inventory: Inventory? = null

    fun towerId(): UUID = towerIdValue

    fun attach(inventory: Inventory) {
        this.inventory = Objects.requireNonNull(inventory, "inventory")
    }

    override fun getInventory(): Inventory =
        inventory ?: throw IllegalStateException("the GUI inventory has not been attached")
}
